qldbdriver/result.go

```diff
@@ -151,6 +161,12 @@ func (result *BufferedResult) GetTimingInformation() *TimingInformation {
 	return result.metrics.timingInformation
 }
 
+// Err is defined so BufferedResult will formally implement the Result interface.
+// Currently there are no error cases so we just return nil.
+func (result *BufferedResult) Err() error {
```

Comment:
We shouldn't do this. They aren't supposed to have the same API. Add a BufferedResult interface as well.